### PR TITLE
mine task Api

### LINKED ISSUE #375: User tasks not being shown on "/mine" route
### Description
When a user who has already been assigned tasks visits the `/mine` route there are no tasks being displayed even if we get the required data from the `/self` API. 

#### Image For reference: 
image

### PR DESCRIPTION
**Pull request by bhtibrewal**
### Changes Made
wrote RTK Query to fetch Mine Task - https://github.com/Real-Dev-Squad/website-status/pull/515/files#diff-4ac2244e7dd24a92566772fd14cbb639b24f63f52996f94de90d8f31107787c7
DELETE [AppContext ](https://github.com/Real-Dev-Squad/website-status/pull/515/files#diff-10506ad712c776ac51b49798b89304915cfad08bcdc1601b8dfb82a1f288b40a) and made [useEditMode hook ](https://github.com/Real-Dev-Squad/website-status/pull/515/files#diff-8c676a2429787e2e8d69471a8af8757f0b653d7fd70b5d20fd3b17cb73129057)
closes #375

### CODE REVIEW

File: src/components/tasks/card/index.tsx [hunk 1]
```diff
@@ -214,17 +215,14 @@ const Card: FC<Props> = ({
...
     }
 
-    const onEditEnabled = () => {
-        actions.onEditRoute();
-    };
     const EditButton = () => (
         <div className={classNames.editButton} data-testid="edit-button">
             <Image
                 src="/pencil.webp"
                 alt="edit"
```
Comment (vvaibhavdesai) [inline src/components/tasks/card/index.tsx:222]: please fix this text

**Review by pallabez — APPROVED** [review]
LGTM.

File: src/pages/challenges.tsx [hunk 2]
```diff
@@ -29,14 +30,14 @@ const renderCardList = (
 
 const Challenges: FC = () => {
     const [filteredChallenge, setFilteredChallenge] = useState<any>([]);
-    const { data: user } = useGetUserQuery(skipToken);
+    const { data: user, isLoading: isAuthenticating } =
```
Comment (vvaibhavdesai) [inline src/pages/challenges.tsx:33]: raising this concern about having meaningful variable naming although as confirmed by @bhtibrewal there is already an issue raised to fix the changes for `isAuthenticating ` and  `isLoggedIn`  which seem some what same to me.

Also please resolve this conversation if needs to be addressed later

File: src/components/tasks/card/index.tsx (hunk 1 above)
Comment (bhtibrewal) [reply src/components/tasks/card/index.tsx:222]: please suggest what should be alt instead

Comment (vvaibhavdesai) [reply src/components/tasks/card/index.tsx:222]: pencil icon to represent edit button  
link if you want to read about it : [Link](https://accessibility.huit.harvard.edu/describe-content-images)

File: src/pages/mine.tsx [hunk 3]
```diff
@@ -1,74 +1,67 @@
...
+                    onContentChange={undefined}
+                />
+            ))}
+        </>
+    );
 }
 
+const Content = () => {
+    const { data: tasks, error, isLoading } = useGetMineTasksQuery(skipToken);
+    if (isLoading) return <p>Loading...</p>;
```
Comment (vvaibhavdesai) [inline src/pages/mine.tsx:30]: nicely done 🎸

File: src/pages/mine.tsx [hunk 4]
```diff
@@ -1,74 +1,67 @@
...
-                                <a
-                                    href={LOGIN_URL}
-                                    target="_blank"
-                                    rel="noreferrer"
-                                >
-                                    Click here to Login
-                                </a>
-                            </div>
-                        ))}
+                    {isAuthenticating ? (
```
Comment (vvaibhavdesai) [inline src/pages/mine.tsx:49]: raise another refactor PR for this multi conditional check

**Review by vvaibhavdesai — APPROVED** [review]

File: src/pages/mine.tsx (hunk 4 above)
Comment (bhtibrewal) [reply src/pages/mine.tsx:49]: okay

File: src/pages/mine.tsx (hunk 3 above)
Comment (bhtibrewal) [reply src/pages/mine.tsx:30]: thanks

File: src/pages/challenges.tsx (hunk 2 above)
Comment (bhtibrewal) [reply src/pages/challenges.tsx:33]: it is being addressed

File: src/components/tasks/card/index.tsx (hunk 1 above)
Comment (bhtibrewal) [reply src/components/tasks/card/index.tsx]: done

**Review by pallabez — APPROVED** [review]
LGTM

**Review by vvaibhavdesai — APPROVED** [review]

**Pull request merged**